File: src/main/java/com/example/jobs/entity/Company.java
package com.example.jobs.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "companies")
public class Company {

  @Id
  private Long id;

  @Column("api_key")
  private String apiKey;

  private String slug = "";

  private String name = "";

  private String website;

  @Column("logo_url")
  private String logoUrl;

  @Column("total_jobs")
  private Integer totalJobs = 0;

  @Column("avg_salary_all")
  private Double avgSalaryAll;

  @Column("salary_count_all")
  private int salaryCountAll = 0;

  private int clicks = 0;

  private int views = 0;

  @Column("views_7d")
  private int views7d = 0;

}

File: src/main/kotlin/com/example/jobs/entity/Company.kt
package com.example.jobs.entity

import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import org.springframework.data.annotation.Id

@Table(name = "companies")
class Company() {
//    @Transient
//    var isNewRecord: Boolean = true

    @Id
    var id: Long? = null

    @Column("api_key")
    var apiKey: String? = null

    var slug: String = ""

    var name: String = ""

    var website: String? = null

    @Column("logo_url")
    var logoUrl: String? = null

    @Column("total_jobs")
    var totalJobs: Int = 0

    @Column("avg_salary_all")
    var avgSalaryAll: Double? = null

    @Column("salary_count_all")
    var salaryCountAll: Int = 0

    var clicks: Int = 0

    var views: Int = 0

    @Column("views_7d")
    var views7d: Int = 0

    constructor(slug: String) : this() {
        this.slug = slug
    }
}